src/basic/terminal-util.c

```diff
@@ -708,6 +709,57 @@ char *resolve_dev_console(char **active) {
         return tty;
 }
 
+int get_kernel_consoles(char ***consoles) {
+        _cleanup_strv_free_ char **con = NULL;
+        _cleanup_free_ char *active = NULL;
+        const char *tty, *state;
+        size_t len;
+        int ret;
+
+        assert(consoles);
+
+        ret = read_one_line_file("/sys/class/tty/console/active", &active);
+        if (ret < 0)
+                return ret;
+
+        FOREACH_WORD(tty, len, active, state) {
```

Comment:
`FOREACH_WORD` was already discussed in the previous PR (https://github.com/systemd/systemd/pull/1432#discussion_r41867412). `extract_first_word` should be used instead.